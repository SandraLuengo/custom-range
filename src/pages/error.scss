@import url("https://fonts.googleapis.com/css?family=Montserrat:400,400i,700");

.error {
    display: flex;
    justify-content: center;
    align-items: center;
    height: 100vh;
    width: 100vw;
    box-sizing: border-box;
    background-color: #ffffff;
    font-family: "Montserrat", sans-serif;
}

h1,
p {
    color: black;
    font-size: 3em;
    padding: 0;
    margin: 0;
}
p {
    font-size: 1em;
}
#render_error {
    fill: none;
    stroke: #4caf50;
    stroke-width: 3;
    stroke-linecap: round;
    stroke-linejoin: round;
    stroke-miterlimit: 10;
}
