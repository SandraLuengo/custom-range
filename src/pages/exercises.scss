.exercises {
  width: 100%;
  height: 100vh;
  display: flex;
  flex-direction: column;
  justify-content: space-around;
  align-items: center;
}

.exercises--btn {
  background-color: #4caf50;
  border: none;
  color: white;
  padding: 8px 20px;
  text-align: center;
  text-decoration: none;
  display: inline-block;
  font-size: 16px;
  margin: 4px 2px;
  cursor: pointer;
  border-radius: 4px;
  a {
    text-decoration: none;
    color: white;
  }
}
