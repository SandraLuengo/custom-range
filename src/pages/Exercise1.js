import React from "react";
import { Link } from "react-router-dom";
import { Range } from "../components";
import "./exercises.scss";

const Exercise1 = () => {
  return (
    <div className="exercises">
      Exercise1
      <Range minPrice={6} maxPrice={80} selectorWidth={10}/>
      <button>
        <Link to={"/"}>Home</Link>
      </button>
    </div>
  );
};

export default Exercise1;
