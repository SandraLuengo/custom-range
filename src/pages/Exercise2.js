import React, { useState, useEffect } from "react";
import { Link } from "react-router-dom";
import { Range, Loading } from "../components";
import useApi from "../hooks/useApi.js"
import "./exercises.scss";

const Exercise2 = () => {

  const { data, error } = useApi("http://demo0922089.mockable.io/exercise2");
  return (
    <div className="exercises">
      {error && <Redirect to="/error"/>}
      <h1>Fixed values range</h1>
      {data ?  <Range
        minPrice={data.min}
        maxPrice={data.max}
        priceArray={data.prices}
        fixedType={true}
      /> : <Loading/> }
      <button className="exercises--btn">
        <Link to={"/"}>Menu</Link>
      </button>
    </div>
  );
};

export default Exercise2;
