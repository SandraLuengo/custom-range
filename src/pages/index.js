import Exercise1 from "./Exercise1.js";
import Exercise2 from "./Exercise2.js";
import Error from "./Error.js";

export { 
    Exercise1, 
    Exercise2,
    Error
 };
