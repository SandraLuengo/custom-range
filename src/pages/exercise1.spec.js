import React from "react";
import Enzyme, { shallow } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
Enzyme.configure({ adapter: new Adapter() });

import { Exercise1 } from "../pages";

let component = null;

describe("Exercise1", () => {
  beforeAll(() => {
    component = shallow(<Exercise1 />);
  });

  it("should render Exercise1", () => {
    expect(component).toMatchSnapshot();
  });
  it.skip("should go to home page on link click", () => {
    component.find(".exercises--btn").simulate("click");
    expect(component.find(".exercises")).toHaveLength(0);
  });
});
