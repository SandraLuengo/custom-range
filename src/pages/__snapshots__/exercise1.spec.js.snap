// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Exercise1 should render Exercise1 1`] = `ShallowWrapper {}`;
